src/grammar.coffee

```diff
@@ -698,6 +711,7 @@ operators = [
   ['nonassoc',  '++', '--']
   ['left',      '?']
   ['right',     'UNARY']
+  ['right',     'AWAIT']
```

Comment:
My thinking in placing it there was that unlike `yield`, `await` would exclusively take promises or other A+ objects as arguments. Therefore it made sense to place its precedence ahead of arithmetic operators which would not result in objects. However,  `await` in ES (and CS) can take any type of argument, but it still seems like a rare use case.